### ISSUE TITLE
Changeset publish seems to be ignoring the `publishConfig` settings

### ISSUE THREAD

**Issue by richbachman** [changesets/changesets#782]
## Affected Packages

@trueplan/forecast-components
@trueplan/forecast-icons
@trueplan/forecast-theme

## Problem

Previous to moving to changesets, we were publishing to the Github package registry. Now we're publishing directory to NPM. In doing so, running the `changeset publish` script seems to ignore our `publishConfig` settings in each of package.json files. See example here: https://github.com/trueplan/forecast/blob/1e3bb8953e7957c6f2287975f94310119df2b0eb/packages/components/package.json#L16

So we're now getting a stray `src/index.ts` directory in the NPM packages which breaks out imports, i.e. `import {foo} from `@trueplan/forecast-components` now has to become `import {foo} from `@trueplan/forecast-components/dist`.

## Proposed solution

Basically `changeset publish` would overwrite the `main` setting in package.json based off the `publishConfig` setting.

**Comment by Andarist** [comment]
This feature request proposes adjustments to "packing" process (preparing a publishable tarball) and proposes for us to get involved in that (so we could "rewrite" some `package.json` fields with the ones defined in `publishConfig`).

I'm afraid that we don't have any plans for this. This feels like something out of the scope of this project and something that would probably open a can of worms for us - it would be somewhat hard to replicate all other assumptions about this packing process that people might have, based on their experience with their package managers. At the moment, we almost completely delegate this responsibility to package managers (npm, yarn, pnpm).

That being said - I would love to hear out how you have been using this before and I would like to discuss alternatives. 

1. What were you using before to accomplish this? Which exact tool supports rewriting those specific fields in this way?
2. I see that your regular `package.json#main` is set to `src/index.ts` - so I assume that the goal of the whole thing is to make the development easier~, to "link" your monorepo packages, so a bundler (or a similar) tool can always resolve to your source files. Would you say that this is correct? What tooling are you using there? What needs to resolve to those source files?

**Comment by richbachman** [comment]
@Andarist thanks for the reply. The error/confusion here stems from a bad rollup config on our end. Closing this issue since we resolved the bad config.
